cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java

```diff
@@ -125,7 +129,132 @@ public void testAllTypesEvaluationSuccess() throws Exception {
           "-o", "B=" + bFile.toURI().toString(),
           "-o", "C=" + cFile.toURI().toString(),
           "-o", "D=" + dFile.toURI().toString(),
-          "-n", "10"
+          "-n", "10",
+          "--overwrite-output-for-contexts"
+        };
+
+        SparkCqlEvaluator.main(args);
+
+        // Expected rows per context were derived from inspecting the input data
+        // by hand and counting the unique values for each context's key column.
+        validateOutputCountsAndColumns(patientFile.toURI().toString(), new HashSet<>(Arrays.asList("pat_id", "MeasureAB.cohort")), 100);
+        validateOutputCountsAndColumns(aFile.toURI().toString(), new HashSet<>(Arrays.asList("id_col", "MeasureA.cohort")), 572);
+        validateOutputCountsAndColumns(bFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureB.cohort")), 575);
+        validateOutputCountsAndColumns(cFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureC.cohort")), 600);
+        validateOutputCountsAndColumns(dFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureD.cohort")), 567);
+    }
+
+    private void validateOutputCountsAndColumns(String filename, Set<String> columnNames, int numExpectedRows) {
+        SparkSession sparkSession = initializeSession(Java8API.ENABLED);
```

Comment:
We have a Spark session that is initialized in the @Before method. Do we need to get another one here or should we just be reusing the one that we already have?

Reply:
The spark session  gets closed when we hit the end of SparkCohortEvaluator. Calling  `initializeSession` here makes sure we re-create a session when needed, or snag the cached session.

Reply:
So, maybe a better suggestion here is to wrap your initializeSession in a try-with-close block? That was my first thought before I remembered we had the annotated methods doing initialization and teardown. It probably doesn't matter much in the end because these are unit tests, but I try to make sure to always close resources when I'm done.